### PR TITLE
[BEEEP] Use MsSqlMigratorUtility for local development databases

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Use the MsSqlMigratorUtility to run local dev database migrations instead of our handcrafted migration code. This is way faster and less to maintain.

## Code changes

* `migrate.ps1` - add new migration code here (it's short enough that it doesn't need a separate file)
* `run_migrations.sh` - deleted as this is replaced with MsSqlMigratorUtility
* `migrate_migration_record.ps1` and `migrate_migrations.sh` - these facilitated the migration to our current scripts, but that was years ago, they're no longer in use. Updated them to facilitate the migration to MsSqlMigratorUtility instead.
* `test-database.yml` - update Github workflow to work with the new script (manual run here: https://github.com/bitwarden/server/actions/runs/8106505885)

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: dev/helpers/mssql/migrate_migrations.sh [hunk 1]
```diff
@@ -16,58 +16,33 @@ while getopts "s" arg; do
...
+    CREATE TABLE [$DATABASE].[dbo].[Migration](
+        [Id] [int] IDENTITY(1,1) NOT NULL,
+        [ScriptName] [nvarchar](255) NOT NULL,
+        [Applied] [datetime] NOT NULL
+    ) ON [PRIMARY];
+
+    ALTER TABLE [$DATABASE].[dbo].[Migration] ADD  CONSTRAINT [PK_Migration_Id] PRIMARY KEY CLUSTERED
+    (
+        [Id] ASC
+    )WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, IGNORE_DUP_KEY = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY];
```
Comment (eliykat) [inline dev/helpers/mssql/migrate_migrations.sh:39]: This is what Azure Data Studio gives you if you right-click the Migration table created by MsSqlMigratorUtility and click "script as CREATE".

File: dev/migrate.ps1 [hunk 2]
```diff
@@ -2,20 +2,21 @@
...
+#  the mssql-tools container which runs under x86_64.
 
 param(
-  [switch]$all = $false,
-  [switch]$postgres = $false,
-  [switch]$mysql = $false,
-  [switch]$mssql = $false,
-  [switch]$sqlite = $false,
-  [switch]$selfhost = $false,
-  [switch]$pipeline = $false
```
Comment (eliykat) [inline dev/migrate.ps1:16]: This is actually not valid syntax and is also not necessary. Switch parameters default to false if not present.

File: dev/migrate.ps1 [hunk 3]
```diff
@@ -29,22 +30,31 @@ if ($all -or $postgres -or $mysql -or $sqlite) {
...
+    $msSqlConnectionString = $(Get-UserSecrets).'dev:selfHostOverride:globalSettings:sqlServer:connectionString'
+    $envName = "self-host"
+
+    Write-Output "Migrating your migrations to use MsSqlMigratorUtility (if needed)"
+    ./migrate_migration_record.ps1 -s
   } elseif ($pipeline) {
-    $migrationArgs = "-p"
+    # pipeline sets this through an environment variable, see test-database.yml
+    $msSqlConnectionString = "$Env:CONN_STR"
+    $envName = "pipeline"
```
[the remark this answers is not in the document]
Comment (eliykat) [reply dev/migrate.ps1]: Good point!

**Comment by eliykat** [comment]
@justindbaur can you please help getting the CI tests working? I keep getting this error:

```
info: Bit.Migrator.DbMigrator[12482444]
      Migrating database.
Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 35 - An internal exception was caught)
 ---> System.Security.Authentication.AuthenticationException: The remote certificate was rejected by the provided RemoteCertificateValidationCallback.
```

I've tried with `Encrypt=True;TrustServerCertificate=True;`, `Encrypt=False`, and removing those variables entirely, but I get exactly the same error. I'm not sure why it's acting differently to our local dev instances.

**Comment by eliykat** [comment]
A bit of trial and error to get the github workflow going, but here's the passing database tests: https://github.com/bitwarden/server/actions/runs/8150568486

**Review by justindbaur — APPROVED** [review]

**Pull request merged**